taskcluster/docker/linux/Dockerfile

```diff
@@ -0,0 +1,168 @@
+    chown worker:worker /builds/worker && \
+    mkdir /builds/worker/artifacts && \
+    chown worker:worker /builds/worker/artifacts
+
+WORKDIR /builds/worker/
+
+# Configuration
+
+ENV ANDROID_BUILD_TOOLS "28.0.3"
+ENV ANDROID_SDK_VERSION "3859397"
+ENV ANDROID_PLATFORM_VERSION "28"
+
+# Set up the language variables to avoid problems (we run locale-gen later).
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+# Do not use fancy output on taskcluster
+ENV TERM dumb
+
+ENV GRADLE_OPTS -Xmx4096m -Dorg.gradle.daemon=false
+
+# Used to detect in scripts whether we are running on taskcluster
+ENV CI_TASKCLUSTER true
+
+ENV \
+    # Some APT packages like 'tzdata' wait for user input on install by default.
+    # https://stackoverflow.com/questions/44331836/apt-get-install-tzdata-noninteractive
+    DEBIAN_FRONTEND=noninteractive
+
+# System.
+
+RUN apt-get update -qq \
+    && apt-get install -qy --no-install-recommends \
+        # To compile Android stuff.
+        openjdk-8-jdk \
+        git \
+        curl \
+        # Required by symbolstore.py.
+        file \
+        # Will set up the timezone to UTC (?).
+        tzdata \
+        # To install UTF-8 locales.
+        locales \
+        # For `cc` crates; see https://github.com/jwilm/alacritty/issues/1440.
+        # <TODO: Is this still true?>.
+        g++ \
+        # <TODO: Explain why we have this dependency>.
+        clang \
+        python3 \
+        python3-pip \
+        # taskcluster > mohawk > setuptools.
+        python3-setuptools \
+        # Required to extract the Android SDK/NDK.
+        unzip \
+        # Required by tooltool to extract tar.xz archives.
+        xz-utils \
+        # Required to build libs/.
+        make \
+        # Required to build sqlcipher.
+        tclsh \
+        # Required in libs/ by some scripts patching the source they download.
+        patch \
+        # For windows cross-compilation.
+        mingw-w64 \
+        ## NSS build dependencies
+        gyp \
+        ninja-build \
+        zlib1g-dev \
+        # <TODO: Delete p7zip once NSS windows is actually compiled instead of downloaded>.
+        p7zip-full \
+        ## End of NSS build dependencies
+        rsync \
+    && apt-get clean
+
+RUN pip3 install --upgrade pip
+RUN pip3 install \
+    'taskcluster>=4,<5' \
+    pyyaml
+
+# Compile the UTF-8 english locale files (required by Python).
+RUN locale-gen en_US.UTF-8
+
+# Android SDK
+
+RUN mkdir -p /builds/worker/android-sdk
+WORKDIR /builds/worker
+
+ENV ANDROID_HOME /builds/worker/android-sdk
+ENV ANDROID_SDK_HOME /builds/worker/android-sdk
+ENV PATH ${PATH}:${ANDROID_SDK_HOME}/tools:${ANDROID_SDK_HOME}/tools/bin:${ANDROID_SDK_HOME}/platform-tools:/opt/tools:${ANDROID_SDK_HOME}/build-tools/${ANDROID_BUILD_TOOLS}
+
+RUN curl -sfSL --retry 5 --retry-delay 10 https://dl.google.com/android/repository/sdk-tools-linux-${ANDROID_SDK_VERSION}.zip > sdk.zip \
```

Comment:
Follow up: https://bugzilla.mozilla.org/show_bug.cgi?id=1622339

Reply:
Noted, added to my list.